src/sentry/relay/config/metric_extraction.py

```diff
@@ -548,31 +549,31 @@ def _convert_aggregate_and_query_to_metrics(
                 )
 
             metric_specs_and_hashes.append((on_demand_spec.query_hash, metric_spec, spec_version))
-        return metric_specs_and_hashes
-    except ValueError:
-        # raised by validate_sampling_condition or metric_spec lacking "condition"
-        metrics.incr(
-            "on_demand_metrics.invalid_metric_spec",
-            tags={"prefilling": prefilling},
-        )
-        logger.exception(
-            "Invalid on-demand metric spec",
-            extra={
-                "dataset": dataset,
-                "aggregate": aggregate,
-                "query": query,
-                "groupbys": groupbys,
-            },
-        )
+        except UnicodeEncodeError:
```

Comment:
We could keep reporting it during the transition period. Let me know what you prefer.